distribution/lib/Standard/Base/0.0.0-dev/src/Random.enso

```diff
@@ -1,99 +1,307 @@
+       - start_time: The lower bound of the range to pick from (inclusive).
+       - end_time: The upper bound of the range to pick from (inclusive).
+
+       The generated time is at a granularity of 1 second.
+
+       Note: the time range end is inclusive because it is impossible to specify
+       an end time after the last second of the day.
+
+       > Example
+         Generate a random time.
+
+             import Standard.Base.Random.Random
+
+             start = Time_Of_Day.new 8 12 15
+             end = Time_Of_Day.new 9 40 2
+             t = Random.time start end
+    time : Time_Of_Day -> Time_Of_Day -> Time_Of_Day
+    time (start_time:Time_Of_Day=(Time_Of_Day.new 0 0 0)) (end_time:Time_Of_Day=(Time_Of_Day.new 23 59 59)) =
+        Random.default_instance.time start_time end_time
+
+    ## GROUP Random
+       Return a new UUID.
+
+       ! `uuid` does not use the underlying `java.util.Random` instance, and is
+       therefore not affected by the seed.
+
+       > Example
+         Generate a uuid.
+
+             import Standard.Base.Random.Random
+
+             uuid = Random.uuid
+    uuid : Text
+    uuid = Random.default_instance.uuid
+
+    ## GROUP Random
+       Returns a new `Vector` containing a random sample of the input vector,
+       with or without replacement.
+
+       Arguments:
+       - vector: The `Vector` to take items from.
+       - count: The number of items to return.
+       - with_replacement: If false, each value in the `Vector` can only be
+         selected once during a single call; the returned values will contain no
+         duplicates. If true, values from the `Vector` can be returned multiple
+         times and duplicates are possible.
+
+       If `with_replacement` is false, and `count` is equal to or greater than
+       the size of the collection, a random permutation of the collection is
+       returned.
+
+       > Example
+         Sample a vector.
+
+             import Standard.Base.Random.Random
+
+             items = Random.items [1, 2, 3, 4]
+    item : Vector Any -> Integer -> Boolean -> Any
+    item (v:Vector) (count:Integer=1) (with_replacement:Boolean=False) = Random.default_instance.item v count with_replacement
+
+    ## GROUP Random
+       Returns a random permutation of the input `Vector`.
+
+       Arguments:
+       - vector: The `Vector` to permute.
+
+       > Example
+         Permute a vector.
+
+             import Standard.Base.Random.Random
+
+             permuted = Random.permute [1, 2, 3, 4]
+    permute : Vector Any -> Vector Any
+    permute (v:Vector) = Random.default_instance.permute v
 
     ## PRIVATE
-       Serialise to JS_Object
-    to_js_object : JS_Object
-    to_js_object self =
-        JS_Object.from_pairs [["type", "Random_Number_Generator"], ["constructor", "new"]]
+       Return the default instance.
+    default_instance : Random_Generator
+    default_instance = Random_Generator.Value Random_Utils.getRandomSingleton
```

Comment:
Calling `default_instance` twice yields different instances. E.g. `default_instance . should_equal default_instance` is going to fail. Does such behavior deserve a name _"default instance"_?

Reply:
Are you sure? I think they will be equal, until `set_seed` is called when they will indeed stop being equal.

Reply:
> I think they will be equal

Opps, yes you are right they will be `==`, but not `Meta.is_same_object`.